lib/options.js

```diff
@@ -92,7 +92,13 @@ export function normalizeOptions(options) {
     const sourceType = normalizeSourceType(options.sourceType);
     const ranges = options.range === true;
     const locations = options.loc === true;
-    const allowReserved = ecmaVersion === 3 ? "never" : false;
+
+    if (ecmaVersion !== 3 && options.allowReserved) {
+
+        // a value of `false` is intentionally allowed here, so a shared config can overwrite it when needed
+        throw new Error("`allowReserved` is only supported when ecmaVersion is 3");
+    }
+    const allowReserved = ecmaVersion === 3 ? (options.allowReserved || "never") : false;
```

Comment:
Also, Acorn has 3 modes for reserved words:

* `allowReserved: true`: allow variables, allow properties.
* `allowReserved: false`: disallow variables, allow properties.
* `allowReserved: "never"`: disallow variables, disallow properties.

Do we want to support all 3 modes, or just 2 of them (`true` and `"never"`)? In either case, what would be allowed values for Espree's `allowReserved` option?

Reply:
as in the tests, `true/false` is expected value (in ES3). IMHO, other values behavior is undefined - it might be working/not working. Do you think we should throw an error in these cases?

Reply:
I think it's better to validate input and accept only values allowed by Espree, and also to pass to Acorn only values that are documented in Acorn's docs.

Assuming that we want to accept only `true` (Acorn's `true`) or `false` (Acorn's `"never"`), this would by chance allow activating the third mode (Acorn's `false`).

Reply:
I’ll update it to only allow a boolean true or false, good catch.

Reply:
Updated.